File: frontend/types/analysis.ts
export interface AnalysisReport {
  core_arguments: string[]
  argument_analysis: string[]
  critical_questions: string[]
  key_quotes: string[]
}

export interface AnalysisRequest {
  url: string
}

export async function analyzeContent(request: AnalysisRequest): Promise<AnalysisReport> {
  console.log("请求体：", request); // 调试用
  const res = await fetch("http://127.0.0.1:8000/api/v1/analysis", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(request),
  });
  if (!res.ok) {
    throw new Error("API 请求失败");
  }
  return res.json();
}

File: backend/app/schemas/analysis.py
from pydantic import BaseModel, Field
from typing import List

class AnalysisRequest(BaseModel):
    url: str = Field(..., description="微信公众号文章链接")

class AnalysisResponse(BaseModel):
    core_arguments: List[str] = Field(..., description="核心观点提炼")
    argument_analysis: List[str] = Field(..., description="论证过程拆解")
    critical_questions: List[str] = Field(..., description="潜在问题与启发")
    key_quotes: List[str] = Field(..., description="金句摘录")
